tests/solana_tests/call.rs

```diff
@@ -733,3 +733,252 @@ fn pda() {
+        assert_eq!(
+            signers[0],
+            create_program_address(&vm.stack[0].id, &[b"foobar"])
+        );
+        assert_eq!(
+            signers[1],
+            create_program_address(
+                &vm.stack[0].id,
+                &[
+                    b"quinquagintaquadringentilliardth",
+                    b"quinquagintaquadringentillionths"
+                ]
+            )
+        );
+        assert_eq!(
+            signers[2],
+            create_program_address(&vm.stack[0].id, &[b"tares", b"enoki"])
+        );
+    };
+
+    let token = Pubkey(
+        "TokenkegQfeZyiNwAJbNbGKPFXCWuBvf9Ss623VQ5DA"
+            .from_base58()
+            .unwrap()
+            .try_into()
+            .unwrap(),
+    );
+
+    vm.account_data.insert(token.0, AccountState::default());
+    vm.call_params_check.insert(token.clone(), test_args);
+
+    vm.function("test")
+        .accounts(vec![("tokenProgram", token.0), ("systemProgram", [0; 32])])
+        .arguments(&[
+            BorshToken::Array(vec![
+                BorshToken::Bytes(b"foo".to_vec()),
+                BorshToken::Bytes(b"bar".to_vec()),
+            ]),
+            BorshToken::Array(vec![
+                BorshToken::Address(*b"quinquagintaquadringentilliardth"),
+                BorshToken::Address(*b"quinquagintaquadringentillionths"),
+            ]),
+            BorshToken::Array(vec![
+                BorshToken::FixedBytes(b"tares".to_vec()),
+                BorshToken::FixedBytes(b"enoki".to_vec()),
+            ]),
+        ])
+        .call();
+}
+
+#[test]
+fn pda_array_of_array() {
+    // now more dynamic
+    let mut vm = build_solidity(
+        r#"
+            import {AccountMeta} from 'solana';
+
+            contract pda {
+                address constant tokenProgramId = address"TokenkegQfeZyiNwAJbNbGKPFXCWuBvf9Ss623VQ5DA";
+                address constant SYSVAR_RENT_PUBKEY = address"SysvarRent111111111111111111111111111111111";
+
+                function test_bytes(bytes[][] seeds) public {
+                    bytes instr = new bytes(1);
+
+                    instr[0] = 0x95;
+
+                    AccountMeta[1] metas = [
+                        AccountMeta({pubkey: SYSVAR_RENT_PUBKEY, is_writable: false, is_signer: false})
+                    ];
+
+                    tokenProgramId.call{seeds: seeds, accounts: metas}(instr);
+                }
+
+                function test_bytes4(bytes4[][] seeds) public {
+                    bytes instr = new bytes(1);
+
+                    instr[0] = 0x95;
+
+                    AccountMeta[1] metas = [
+                        AccountMeta({pubkey: SYSVAR_RENT_PUBKEY, is_writable: false, is_signer: false})
+                    ];
+
+                    tokenProgramId.call{seeds: seeds, accounts: metas}(instr);
+                }
+
+                function test_addr(address[][] seeds) public {
```

Comment:
If change the type to `address[][2]`, I get this error:
```
Assertion failed: (Ty && "Invalid GetElementPtrInst indices for type!"), function checkGEPType, file Instructions.h, line 922.
zsh: abort      ./target/debug/solang compile --target solana ../solang/examples/test.sol
```

If the type is `address[2][]`, the error is:
```
thread 'main' panicked at 'Found ArrayValue(ArrayValue { name: "elem21", address: 0x6000011e1820, is_const: false, is_const_array: false, is_const_data_array: false, is_null: false, llvm_value: "  %elem21 = load [2 x [32 x i8]], ptr %index_access20, align 1", llvm_type: ArrayType { array_type: Type { address: 0x15b075078, llvm_type: "[2 x [32 x i8]]" } } }) but expected PointerValue variant', /Users/lucasste/.cargo/registry/src/index.crates.io-6f17d22bba15001f/inkwell-0.2.0/src/values/enums.rs:302:13
```

Now, if it is `address[2][4]`, the error is:
```
thread 'main' panicked at 'Found ArrayValue(ArrayValue { name: "elem17", address: 0x6000002a54c0, is_const: false, is_const_array: false, is_const_data_array: false, is_null: false, llvm_value: "  %elem17 = load [2 x [32 x i8]], ptr %index_access16, align 1", llvm_type: ArrayType { array_type: Type { address: 0x15485be78, llvm_type: "[2 x [32 x i8]]" } } }) but expected PointerValue variant', /Users/lucasste/.cargo/registry/src/index.crates.io-6f17d22bba15001f/inkwell-0.2.0/src/values/enums.rs:302:13
```

These errors go away when I comment the external call.

Reply:
You're right, I forgot to test fixed length arrays.